drivers/SmartThings/philips-hue/src/utils.lua

```diff
@@ -5,6 +5,11 @@ local utils = {}
 
 local MAC_ADDRESS_STR_LEN = 12
 
+
+function utils.kelvin_to_mirek(kelvin) return 1000000 / kelvin end
```

Comment:
isn't the unit a "mired"?

and I don't know if hue is better-behaved, but we've definitely run into situations where we get divby0 errors here before on the zigbee side

Reply:
> isn't the unit a "mired"?

You and I actually had this conversation when I first wrote the driver a few years ago.

They're the same unit with different names. Mired is _**micro reciprocal degree**_, an acronym. Mirek is the SI Unit, but nobody uses it. [From Wikipedia](https://en.wikipedia.org/wiki/Mired):

```
The [SI](https://en.wikipedia.org/wiki/SI) term for this unit is the reciprocal megakelvin (MK−1), shortened to mirek, but this term has not gained traction.[[1]](https://en.wikipedia.org/wiki/Mired#cite_note-1)
```

The Hue API and documentation use mirek, so I’m using that for consistency here.

> and I don't know if hue is better-behaved, but we've definitely run into situations where we get divby0 errors here before on the zigbee side

We sanitize the input kelvin for capability commands and the input mireks for attribute emits before calling this function:

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1335/files#diff-ddccea1fd0c65608f45092f0f576c125a2afaebd37e7b9bb78855eece84037e5R202